src/controller/CHIPDeviceController.cpp

```diff
@@ -2513,19 +2516,20 @@ CHIP_ERROR
 DeviceCommissioner::SendCommissioningCommand(DeviceProxy * device, const RequestObjectT & request,
                                              CommandResponseSuccessCallback<typename RequestObjectT::ResponseType> successCb,
                                              CommandResponseFailureCallback failureCb, EndpointId endpoint,
-                                             Optional<System::Clock::Timeout> timeout)
+                                             Optional<System::Clock::Timeout> timeout, bool useContext)
 
 {
-    VerifyOrDie(!mInvokeCancelFn); // we don't make parallel calls
+    VerifyOrDie(!(useContext && mInvokeCancelFn)); // we don't make parallel calls that we track with context
 
-    auto onSuccessCb = [context = this, successCb](const app::ConcreteCommandPath & aPath, const app::StatusIB & aStatus,
-                                                   const typename RequestObjectT::ResponseType & responseData) {
+    void * context   = (useContext) ? this : nullptr;
```

Comment:
Why the extra parens?

Reply:
I think ternary expressions are more readable when the condition is in parentheses.